tfsdk/serve_import_test.go

```diff
@@ -0,0 +1,181 @@
+package tfsdk
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+)
+
+func TestServerImportResourceState(t *testing.T) {
+	t.Parallel()
+
+	type testCase struct {
+		req *tfprotov6.ImportResourceStateRequest
+
+		impl func(context.Context, ImportResourceStateRequest, *ImportResourceStateResponse)
+
+		resp *tfprotov6.ImportResourceStateResponse
+	}
+
+	tests := map[string]testCase{
+		"Set": {
+			req: &tfprotov6.ImportResourceStateRequest{
+				ID:       "test",
+				TypeName: "test_import_state",
+			},
+
+			impl: func(ctx context.Context, req ImportResourceStateRequest, resp *ImportResourceStateResponse) {
+				state := testServeResourceImportStateData{
+					Id: req.ID,
+				}
+
+				diags := resp.State.Set(ctx, state)
+				resp.Diagnostics.Append(diags...)
+			},
+			resp: &tfprotov6.ImportResourceStateResponse{
+				ImportedResources: []*tfprotov6.ImportedResource{
+					{
+						State: func() *tfprotov6.DynamicValue {
+							val, err := tfprotov6.NewDynamicValue(
+								testServeResourceTypeImportStateTftype,
+								tftypes.NewValue(
+									testServeResourceTypeImportStateTftype,
+									map[string]tftypes.Value{
+										"id":              tftypes.NewValue(tftypes.String, "test"),
+										"optional_string": tftypes.NewValue(tftypes.String, nil),
+										"required_string": tftypes.NewValue(tftypes.String, ""),
+									},
+								),
+							)
+							if err != nil {
+								panic(err)
+							}
+							return &val
+						}(),
+						TypeName: "test_import_state",
+					},
+				},
+			},
+		},
+		"SetAttribute": {
+			req: &tfprotov6.ImportResourceStateRequest{
+				ID:       "test",
+				TypeName: "test_import_state",
+			},
+
+			impl: func(ctx context.Context, req ImportResourceStateRequest, resp *ImportResourceStateResponse) {
+				diags := resp.State.SetAttribute(ctx, tftypes.NewAttributePath().WithAttributeName("id"), req.ID)
+				resp.Diagnostics.Append(diags...)
+			},
+
+			resp: &tfprotov6.ImportResourceStateResponse{
+				// TODO: SetAttribute should error or import should do the right thing here as
+				//       this seems like a potentially common implementation.
+				// Reference: https://github.com/hashicorp/terraform-plugin-framework/issues/148
```

Comment:
If this is not going to be resolved prior to v0.4.0, I suggest we at least update the ImportState method Go documentation to note that using resp.State.Set() is required for now.

Reply:
I'd say we should prioritize it for 0.4.0.